### PR TITLE
Merge 1.0.0 beta.10.3 proposed update into 1.0.0

### LINKED ISSUE #1186: Prepare beta-10.3 release
**What Needs to be Done?**

Do it.

### PR DESCRIPTION
**Pull request by lifflander**
Fixes #1186, #1194

### CODE REVIEW

**Comment by cz4rs** [comment]
ToDo: 

- [ ] #728 
- [ ] #1111 
- [ ] #1163 
- [ ] #1183

**Comment by cz4rs** [comment]
- builds pass cleanly at 7290bec (skipping quality review):
image

- merge commits are missing, but the history is cleaned-up and linear
- I am currently working on backporting #728 - main problem is that there are some bits from #796 involved (overall, I am trying to backport Phil's solution while maintaning the `reduce` interface from `1.0.0`)

**Comment by lifflander** [comment]
> * builds pass cleanly at 7290bec](https://github.com/DARMA-tasking/vt/commit/7290bec2be4c189158979638355bce3bb0be3372) (skipping quality review):
> * merge commits are missing, but the history is cleaned-up and linear
> * I am currently working on backporting #728 - main problem is that there are some bits from #796 involved (overall, I am trying to backport Phil's solution while maintaning the `reduce` interface from `1.0.0`)

Looks good so far. Let me know if you need any help from me. The reduce interface did change a lot on develop, but I think the changes Phil made should generally be applicable to 1.0.0.

**Comment by cz4rs** [comment]
pushed single commit from #908 to fix another Intel 19 issue (95ff3ea), it might be necessary to include complete solution

ToDo:
- [ ] add nvcc builds to CI and fix any remaining issues
- [ ] add #908 to the release (?)

**Comment by cz4rs** [comment]
@lifflander 
- I have added up-to-date build definitions for nvcc
- all the essential builds are passing (including intel 19, nvidia 10.1, nvidia 11.0)

File: src/vt/messaging/active.cc [hunk 1]
```diff
@@ -189,21 +310,13 @@ EventType ActiveMessenger::sendMsgBytes(
...
-  auto mpi_event = holder.get_event();
-
   if (!is_term || backend_check_enabled(print_term_msgs)) {
     debug_print(
       active, node,
       "sendMsgBytes: size={}, dest={}\n", msg_size, dest
     );
   }
 
-  if (is_shared) {
```
Comment (PhilMiller) [inline src/vt/messaging/active.cc:203]: This was apparently the only use of `is_shared`, which GCC with `-Werror=unused-variable` now complains about.

I'm going to open an issue and PR for that, which can be cherry-picked here.

Comment (PhilMiller) [reply src/vt/messaging/active.cc:203]: It seems this only exists on the release branch. With the variable deleted, EMPIRE builds. I'm running the tests now.

**Review by PhilMiller — APPROVED** [review]
The code changes look as expected, and EMPIRE tests pass.

**Comment by PhilMiller** [comment]
> The code changes look as expected, and EMPIRE tests pass.

Those tests were with GCC 7.2.0. I'm building and testing with Cuda 9.2 now.

**Comment by PhilMiller** [comment]
> > The code changes look as expected, and EMPIRE tests pass.
> 
> Those tests were with GCC 7.2.0. I'm building and testing with Cuda 9.2 now.

And, they passed.

I can merge this, unless there's something else that needs to happen with it first

**Pull request merged**